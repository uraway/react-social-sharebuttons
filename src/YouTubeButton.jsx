// @flow
import React, { Component } from 'react';

type Props = {
  channel: string,
  layout?: 'none' | 'full',
  theme?: 'none' | 'dark',
  count?: 'show' | 'hidden',
};

export default class YouTubeButton extends Component<Props> {
  node = null;

  componentDidMount() {
    const s = document.createElement('script');
    s.src = 'https://apis.google.com/js/platform.js';
    if (this.node && this.node.parentNode) this.node.parentNode.appendChild(s);
  }

  render() {
    const { channel, layout, theme, count } = this.props;
    return (
      <div>
        <div
          ref={(node) => (this.node = node)}
          className="g-ytsubscribe"
          data-channel={channel}
          data-layout={layout}
          data-theme={theme}
          data-count={count}
        />
      </div>
    );
  }
}

YouTubeButton.defaultProps = { layout: 'none', theme: 'none', count: 'show' };
