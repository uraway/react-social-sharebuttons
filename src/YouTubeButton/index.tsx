import { FC } from 'react';
import { useLayoutEffectOnce } from '../utils';

export type YouTubeButtonProps = {
  channel: string;
  layout?: 'none' | 'full';
  theme?: 'none' | 'dark';
  count?: 'show' | 'hidden';
};

const YouTubeButton: FC<YouTubeButtonProps> = (props) => {
  useLayoutEffectOnce(() => {
    const script = document.createElement('script');
    script.src = 'https://apis.google.com/js/platform.js';
    document.getElementsByTagName('head')[0]?.appendChild(script);

    return () => script.remove();
  });

  const { channel, layout, theme, count } = props;
  return (
    <div>
      <div
        className="g-ytsubscribe"
        data-channel={channel}
        data-layout={layout}
        data-theme={theme}
        data-count={count}
      />
    </div>
  );
};

YouTubeButton.defaultProps = { layout: 'none', theme: 'none', count: 'show' };

export default YouTubeButton;
