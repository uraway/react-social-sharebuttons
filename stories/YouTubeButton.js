import { text, select } from '@storybook/addon-knobs';
import { YouTubeButton } from '../dist/index.esm.js';

export default {
  title: 'YouTubeButton',
};

const layoutOptions = {
  none: 'none',
  full: 'full',
};

const themeOptions = {
  none: 'none',
  dark: 'dark',
};

const countOptions = {
  show: 'show',
  hidden: 'hidden',
};

export const Basic = () => (
  <YouTubeButton
    channel={text('channel', 'GoogleDevelopers')}
    layout={select('layout', layoutOptions)}
    theme={select('theme', themeOptions)}
    count={select('count', countOptions)}
  />
);
