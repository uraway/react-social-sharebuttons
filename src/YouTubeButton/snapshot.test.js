import React from 'react';
import renderer from 'react-test-renderer';
import YouTubeButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<YouTubeButton channel="channel" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<YouTubeButton channel="channel" layout="full" theme="dark" count="hidden" />).toJSON();
  expect(tree).toMatchSnapshot();
});
