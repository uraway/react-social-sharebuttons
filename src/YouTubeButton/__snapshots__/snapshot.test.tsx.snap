// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div
  className="g-ytsubscribe"
  data-channel="channel"
  data-count="hidden"
  data-layout="full"
  data-theme="dark"
/>
`;

exports[`renders correctly with required props 1`] = `
<div
  className="g-ytsubscribe"
  data-channel="channel"
  data-count="show"
  data-layout="none"
  data-theme="none"
/>
`;
